pkg/importer/dataStream.go

```diff
@@ -973,35 +970,34 @@ func (d *DataStream) copy(dest string) (err error) {
 //Returns failure if either of the following happens
 //1. info cannot be retrieved from qcow2 file
 //2. Either ActualSize or VirtualSize are 0 - not specified
-//3. ActualSize and VirtualSize together exceed available PVC Space
-func ValidateSpaceConstraint(qcow2Image string, destDir string) (bool, error) {
+//3. The current available space < virtual size
+func ValidateSpaceConstraint(qcow2Image string, destDir string) error {
 
 	//Verify there is enough space in pvc before conversion
 	info, err := qemuOperations.Info(qcow2Image)
 
+	// Failed to get info on source image.
 	if err != nil {
-		return false, errors.Wrap(err, "Local image validation failed to retrieve image info")
-	}
-
-	if info == nil {
-		return false, errors.Wrap(err, "Local image validation failed to retrieve image info")
+		return errors.Wrap(err, "Local image validation failed to retrieve image info")
 	}
 
+	// Got some info, but couldn't get virtual or actual size.
 	if info.VirtualSize == 0 || info.ActualSize == 0 {
-		return false, errors.Wrap(err, "Local image validation failed - no image size info is provided")
+		return errors.New("Local image validation failed - no image size info is provided")
 	}
 
-	convRequiredSpace := info.VirtualSize + info.ActualSize
-
+	// The converted file is guaranteed not the be larger than the virtual size, so we need to compare current available space (which is total space
+	// - temporary file actual size) against the worst case scenario which is the conversion actual size equals the virtual size.
+	convRequiredSpace := info.VirtualSize
```

Comment:
Can we assume that this function will always be called when the qcow2 file has been downloaded to the pvc?  For example, it may make sense to call this before the qcow2 file is downloaded.  Which we could do with http qcow2.

Reply:
Yes we can, this function is called from convertQcow2ToRaw, which is only called in the copy function, in the path where we are guaranteed that that the temporary file has been copied to the temporary location. Then convertQcow2ToRaw is called to convert the existing temporary file to the target format in the target location. This method is not called in the streaming path.

Reply:
I understand that this is the case now, but wouldn't it be better if we could validate before we downloaded?  And we can in some cases.  I'm not saying we should do that now.  But maybe it would be better if this function could somehow handle both cases by adding an additional arg or something.

Reply:
Well I would prefer if this whole method simply went away, and it can with the scratch space initiative. Since we will store the temporary files in the scratch space, this whole calculation is no longer needed at that point. And I just pushed the first WIP PR for scratch space #659

Reply:
Okay, we'll see how this works out in your PR.